resources/actions/nexus.py

```diff
@@ -0,0 +1,330 @@
+                },
+          "cleanup": {
+                  "policyNames": [
+                            "string"
+                          ]
+                },
+          "component": {
+                  "proprietaryComponents": "false"
+                },
+          "yum": {
+                  "repodataDepth": repo_data_depth,
+                  "deployPolicy": "STRICT"
+                }
+    }
+
+    logging.info("Creating hosted yum repository: %s" % name)
+    response = requests.post(api_url, json=params,
+                             auth=requests.auth.HTTPBasicAuth(user, password))
+
+    if response.status_code == 201:
+        logging.info('Done')
+    else:
+        logging.error('Failed to create repository: %s' % name)
+        logging.error('Status code: %s' % response.status_code)
+        logging.error('Response: %s' % response.content)
+        sys.exit(1)
+
+
+def get_yum_repo(url, name, user=None, password=None):
+    api_url = "%s/service/rest/v1/repositories/yum/hosted/%s" % (url, name)
+
+    auth_params = None
+
+    if user is not None and password is not None:
+        auth_params = requests.auth.HTTPBasicAuth(user, password)
+
+    response = requests.get(api_url, auth=auth_params)
+    if response.status_code == 200:
+        response_json = json.loads(response.content)
+        print(json.dumps(response_json, indent=2))
+    else:
+        logging.error('Failed to get repository: %s' % name)
+        logging.error('Status code: %s' % response.status_code)
+        logging.error('Response: %s' % response.content)
+        sys.exit(1)
+
+
+def create_apt_repo(url, name, user, password,
+                    blob_store, write_policy,
+                    distribution, keypair=None, keypair_file=None,
+                    passphrase=None):
+
+    api_url = "%s/service/rest/v1/repositories/apt/hosted" % url
+
+    if keypair_file:
+        keypair = keypair_file.read()
+
+    params = {
+          "name": name,
+          "online": "true",
+          "storage": {
+            "blobStoreName": blob_store,
+            "strictContentTypeValidation": "true",
+            "writePolicy": write_policy
+          },
+          "cleanup": {
+            "policyNames": [
+              "string"
+            ]
+          },
+          "component": {
+            "proprietaryComponents": "true"
+          },
+          "apt": {
+            "distribution": distribution
+          },
+          "aptSigning": {
+            "keypair": keypair,
+            "passphrase": passphrase
+          }
+    }
+
+    logging.info('Creating hosted APT repository: %s' % name)
+    response = requests.post(api_url, json=params,
+                             auth=requests.auth.HTTPBasicAuth(user, password))
+
+    if response.status_code == 201:
+        logging.info('Done')
+    else:
+        logging.error('Failed to create repository: %s' % name)
+        logging.error('Status code: %s' % response.status_code)
```

Comment:
Using f-strings is faster and recommended way for Python 3.x:
```python
logging.error(f'Status code: {response.status_code}')
```

Reply:
Fixed